financial_transcripts/rag-cosmos-mongo/ChatFlow.py

```diff
@@ -0,0 +1,27 @@
+
+from promptflow import tool
+
+
+# The inputs section will change based on the arguments of the tool function, after you save the code
+# Adding type to arguments and return value will help the system show the types properly
+# Please update the function name/signature per need
+@tool
+def chat_flow(check_existing_context: bool, filter: {}, searchType: str) -> str:
+    if check_existing_context == False and filter != {} and "filter" in searchType:
+        return "new_retrieval"
+    elif check_existing_context == False and filter != {} and "filter" not in searchType:
```

Comment:
Some of the conditions here appear to be redundant. Please consolidate.

Reply:
Well, I believe these were added by journey and might be necessary if I recall correctly.